web/src/main/java/org/springframework/security/web/util/matcher/OrRequestMatcher.java

```diff
@@ -28,17 +28,18 @@
  * {@link RequestMatcher} instances match.
  *
  * @author Rob Winch
+ * @author Evgeniy Cheban
  * @since 3.2
  */
 public final class OrRequestMatcher implements RequestMatcher {
 
-	private final List<RequestMatcher> requestMatchers;
+	private final List<? extends RequestMatcher> requestMatchers;
 
 	/**
 	 * Creates a new instance
 	 * @param requestMatchers the {@link RequestMatcher} instances to try
 	 */
-	public OrRequestMatcher(List<RequestMatcher> requestMatchers) {
+	public OrRequestMatcher(List<? extends RequestMatcher> requestMatchers) {
```

Comment:
I'd recommend trying to keep the matchers the same as they are today. It might mean registering more authorization manager instances in the configurer.